plugins/modules/sonic_image_management.py

```diff
@@ -351,6 +399,16 @@ def execute_command(module, params, result):
                 payload['openconfig-image-management:input'] = {'image-name': params['path']}
             elif (params['command'] == 'remove' and params['name'] != 'all') or params['command'] == 'set-default':
                 payload['openconfig-image-management:input'] = {'image-name': params['name']}
+            elif params['command'] == 'gpg-key':
+                payload['openconfig-image-management:input'] = {"key-server": params['keyserver'], "key-id": params['pubkeyid']}
+            elif params['command'] == 'verify':
+                if params['verifymethod'] == 'gpg':
+                    payload['openconfig-image-management:input'] = {
+                        "image-name": params['name'], "verify-method": params['verifymethod'], "sigfilename": params['signaturefile']}
+                else:
+                    payload['openconfig-image-management:input'] = {
+                        "image-name": params['name'], "verify-method": params['verifymethod'],
+                        "sigfilename": params['signaturefile'], "keyfilename": params['pubkeyfilename']}
```

Comment:
Could be simplified
```suggestion
                payload['openconfig-image-management:input'] = {
                        "image-name": params['name'], "verify-method": params['verifymethod'], "sigfilename": params['signaturefile']
                }
                if params['verifymethod'] == 'pki':
                    payload['openconfig-image-management:input']['keyfilename'] = params['pubkeyfilename']
```

Reply:
It is better to keep this check. The suggested change would initially initialize for both GPG and PKI and will be overridden for PKI.
The existing method checks for GPG and PKI separately and assigns payload accordingly.